src/citrine/_rest/engine_resource.py

```diff
@@ -53,13 +45,31 @@ def _post_dump(self, data: dict) -> dict:
         # Only the data portion of an entity is sent to the server.
         data = data["data"]
 
-        # Currently, name and description exists on both the data envelope and the config.
-        data["instance"]["name"] = data["name"]
-        data["instance"]["description"] = data["description"]
+        if "instance" in data:
+            # Currently, name and description exists on both the data envelope and the config.
+            data["instance"]["name"] = data["name"]
+            data["instance"]["description"] = data["description"]
 
         return super()._post_dump(data)
 
 
+# This class is the base type for the new module endpoints which do not support versions. If/once
+# they support versioning, they should be switched to inherit from VersionedEngineResource.
+class EngineResource(EngineResourceWithoutStatus[Self], IncludeParentProperties[Self]):
+    """Base resource for metadata from stand-alone AI Engine modules."""
+
+    status = properties.Optional(properties.String(), 'metadata.status.name', serializable=False)
+    """:Optional[str]: short description of the resource's status"""
+    status_detail = properties.List(properties.Object(StatusDetail), 'metadata.status.detail',
+                                    default=[], serializable=False)
+    """:List[StatusDetail]: a list of structured status info, containing the message and level"""
+
+    @classmethod
+    def build(cls, data: dict):
+        """Build an instance of this object from given data."""
+        return super().build_with_parent(data, __class__)
```

Comment:
I don't know how we feel about making significant improvements to the properties module, but this seems like something that would make sense to make core.

Reply:
Our intent to replace it is part of why I never bothered. If I end up stalling out on Pydantic, then maybe it becomes worthwhile.

Reply:
It's amazing how code collapse can make you blind...